src/singpass-button/singpass-button.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { getSpLogo } from "./singpass-assets";
+import { Main, MainStyleProps } from "./singpass-button.style";
+import { SingpassButtonProps, SingpassButtonRef } from "./types";
+import { Typography } from "../typography";
+
+/**
+ * NOTE: Due to the way we intend to customise both components, with forwardRef behaviour
+ * we are unable to create a single component and have them share.
+ *
+ * Will refactor if there is a better way
+ */
+const DefaultComponent = (
+    props: SingpassButtonProps,
+    ref: SingpassButtonRef
+) => {
+    const {
+        styleType = "white-filled",
+        singpassLogoStyleOverride,
+        ...otherProps
+    } = props;
+
+    const mainStyle: MainStyleProps = {
+        $buttonStyle: styleType,
+        $buttonSizeStyle: "default",
+        $singpassLogoOverride: singpassLogoStyleOverride,
+    };
+
+    return (
+        <Main
+            ref={ref}
+            data-testid={otherProps["data-testid"] || "button"}
+            {...mainStyle}
+            {...otherProps}
+            aria-label="Log in with sing pass"
+        >
+            <Typography.HeadingXS inline weight="semibold">
+                Log in with
+            </Typography.HeadingXS>
```

Comment:
use the `Font` token in the css instead

and I think it should be bold based on the examples in the guidelines? what does your product Figma use?

- body-md-bold
- heading-md-bold
- heading-xs-bold